### ISSUE TITLE
Standalone blog broken on Insiders

### ISSUE THREAD

**Issue by StarfallProjects** [squidfunk/mkdocs-material#5922]
### Bug description

I am using Insiders to build a standalone blog site. I just realised my latest builds aren't actually generating the blog: the blog posts aren't getting added to the nav or displayed on the site.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Setting up blog](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/)

### Reproduction

[9.2.5+insiders.4.40.1-insiders-blog-borked.zip](https://github.com/squidfunk/mkdocs-material/files/12470003/9.2.5%2Binsiders.4.40.1-insiders-blog-borked.zip)

### Steps to reproduce

1. Create a fresh project and fresh virtual env, per the reproduction guide
2. Install the latest Insider with `pip install git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git` (replace token with your token)
3. Set the site up to be a standalone blog site:
```
plugins:
  - blog:
      blog_dir: .
```

My full mkdocs.yml:

```
site_name: My Docs
theme:
  name: material
plugins:
  - info
  - blog:
      blog_dir: .
nav:
  - index.md
```

4. Run locally with `mkdocs serve`

Result is the site builds. It warns you the blog posts aren't included in the navigation, and they do not appear on the site. 

Python version 3.11.2 when creating the reproduction, but have seen the issue on 3.10.x as well. Issue has occurred both locally and on my live site built with Cloudflare Pages. Worth noting I updated to a newer Python version as a fix for another issue (which I believe has already been reported here: https://github.com/squidfunk/mkdocs-material/issues/5916)

This issue only appeared recently, but I can't pinpoint when exactly (I have a horrible feeling my blog has been down a couple of days without me noticing) As a temporary fix I replaced Insiders with normal Material, and that's working fine (also worked fine with the minimal reproduction)

The issue only seems to affect standalone blogs. When I changed the minimal repro to use the default blog directory, it worked.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and for the detailed analysis! Looking into it.

**Comment by squidfunk** [comment]
Fixed in [`a0280cb`](https://github.com/squidfunk/mkdocs-material-insiders/commit/a0280cb83dc0bf11af3aa6f6df17d0615eb1ab1f). I've taken the time and did a refactoring, unifying how views and pages are constructed. There might still be issues, as it's still a ground up rewrite of what we had before, but I feel that we made a big leap in the right direction!

**Comment by squidfunk** [comment]
The problem was directly related to an attempt to fix custom index pages, more specifically:

- #5896 
- #5904

Those should now work again as expected. I'm still finishing the last tests and will issue a new release later today.

**Comment by squidfunk** [comment]
Released as part of [9.2.6+insiders-4.40.2](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.40.2).
